pkg/sql/catalog/descs/collection.go

```diff
@@ -269,6 +269,20 @@ func (tc *Collection) WriteDescToBatch(
 			return err
 		}
 	}
+	// Retrieve the expected bytes of `desc` in storage.
+	// If this is the first time we write to `desc` in the transaction, its
+	// expected bytes will be retrieved when we read it into this desc.Collection,
+	// and carried over in it.
+	// If, however, this is not the first time we write to `desc` in the transaction,
+	// which means it has existed in `tc.uncommitted`, we will retrieve the expected
+	// bytes from there.
+	var expected []byte
+	if exist := tc.uncommitted.uncommitted.GetByID(desc.GetID()); exist == nil {
```

Comment:
nit: use `getUncommittedByID` here instead

uber nit: perhaps it's just me but evaluating the `nil` case first in this if-block feels weird to me.